### DISCUSSION TITLE
VSC's launch.json for the minimal Extension example

### DISCUSSION THREAD

**Discussion by weclicktop** [DevToys-app/DevToys#1383] [Q&A]
Hi there,

I am trying to work thru your extension development doc step by step. When I go to the [Build & Run](https://devtoys.app/doc/articles/extension-development/getting-started/debug-an-extension.html#build--run) part, it looks to me that I need to have a 'launch.json' file ready before I can have it run by clicking F5 in VS code. Could you provide a sample launch.json file as well?

Thanks for your help!

**Comment by veler** [comment]
Hello,
Do you have [C# Dev Kit](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csdevkit) extension installed in VS Code? You'd need this with the `launchSettings.json` (not `launch.json`).

**Reply by weclicktop** [reply]
Yes it is installed. When I hit F5, C# Dev Kit will complain "No C# project is currently loaded. Please create a C# project in order to debug."

**Reply by veler** [reply]
Gotcha. Do you have a CSPROJ file? Is it loaded in VS Code (you can use command palette to open a .NET solution or project)? What if you do a right click on the project node in C# DevKit's Solution Explorer, and Start Debugging?

**Reply by weclicktop** [reply]
Thanks! CSPROJ file is created as well. I can open the .NET solution thru command palette. When I follow your "right click on project node and Start Debugging", C# Devkit will complain "Unable to convert launch target to a vscode debug configuration."

**Reply by weclicktop** [reply]
I have also followed your steps in Visual Studio 2022, everything looks good. 

Attached please find my testing code in VSC.
[toy_test.zip](https://github.com/user-attachments/files/16647899/toy_test.zip)

**Reply by veler** [reply]
Thanks for the follow up. I'm able to reproduce the issue now, and try to understand what's going on :)
